enhancements/machine-config/manage-boot-images.md

```diff
@@ -193,6 +194,8 @@ Based on the observation above, here is a rough outline of what CAPI support wou
 
 Much of the existing design regarding architecture & platform detection, opt-in, degradation and storing boot image history can remain the same. 
 
+When [MachineDeployments](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine-deployment#machinedeployment) are introduced into CAPI, this mechanism will need to be reworked to update those rather than the MachineSet itself.
```

Comment:
I'm wondering if we will need the MCO to recognise that the MachineSet has a parent and therefore not to modify the object when it's owned by something that would otherwise manage this property

Reply:
Hmm, I could put in a condition for this so we don't accidentally take over management from a parent. Is this typically an annotation/label on the machineset that tells me who owns it? I was hoping the opt-in step would prevent cases of this happening.

Reply:
Why wouldn't we just act specifically on the MachineDeployments?

Reply:
We are expecting in the future that some clusters will have MachineSets, some with have MachineDeployments, and some will have a mixture of the two. So I would expect this mechanism to be able to cope with both. 

I guess the selection that users make via the API could express which resources need to be updated, rather than it being baked in here as additional logic

Reply:
Coming back to this comment as well, how does this work with Hive, Hive will spec machinesets and expect their content to be a specific value, this may interfere with the Hive mechanisms

>  Is this typically an annotation/label on the machineset that tells me who owns it?

It would be an `OwnerReference` on the metadata of the object

Reply:
Understood, will add this an additional guard on machinesets on my next pass. Thanks!

Reply:
+1 on this CAPI MachineSet vs CAPI MachineDeployment story. It will be important to support both in the MSBIC in the future. So planning for this ahead would not be a bad idea.